modules/fieldlinessequence/rendering/renderablefieldlinessequence.cpp

```diff
@@ -54,243 +47,294 @@ namespace {
+
+        // Set if first/last file should render forever
+        bool showAtAllTimes;
+
+        // If using dynamic downloading, files are not cached by default.
+        // To save files locally , set this to true.
+        std::optional<bool> CacheData;
+
         // If data sets parameter start_time differ from start of run,
         // elapsed_time_in_seconds might be in relation to start of run.
         // ManuelTimeOffset will be added to trigger time.
-        std::optional<double> manualTimeOffset;
+        std::optional<float> manualTimeOffset;
+
+        enum class [[codegen::map(openspace::fls::Model)]] Model {
+            Batsrus,
+            Enlil,
+            Pfss
+        };
+        // Currently supports: batsrus, enlil & pfss. Not specified -> model == invalid
+        // which just means that the scaleFactor (scaleToMeters) will be 1.f assuming
+        // meter as input
+        std::optional<Model> simulationModel;
+
+        // Convert the models distance unit, ex. AU to meters for Enlil.
+        // 1.f is default, assuming meters as input.
+        // Does not need to be specified if simulationModel is specified.
+        // Using a different model? Set this to scale your vertex positions to meters.
+        std::optional<float> scaleToMeters;
+
+        // choose type of loading:
+        //0: static loading and static downloading
+        //1: dynamic loading and dynamic downloading
+        enum class [[codegen::map(openspace::RenderableFieldlinesSequence::LoadingType)]]
+        LoadingType {
+            StaticLoading,
+            DynamicDownloading
+        };
+        std::optional<LoadingType> loadingType;
+        // dataID that corresponds to what dataset to use if using dynamicWebContent
+        std::optional<int> dataID;
+        // number Of Files To Queue is a max value of the amount of files to queue up
+        // so that not to big of a data set is downloaded nessesarily.
+        std::optional<int> numberOfFilesToQueue;
+        std::optional<std::string> infoURL;
+        std::optional<std::string> dataURL;
+
+        enum class [[codegen::map(openspace::RenderableFieldlinesSequence::SourceFileType)]]
+        SourceFileType {
+            Cdf,
+            Json,
+            Osfls
+        };
+        // Specify file type: Cdf, Json or Osfls
+        SourceFileType inputFileType;
+
+        // Path to folder containing the input files
+        std::optional<std::filesystem::path> sourceFolder [[codegen::directory()]];
+        // Path to a .txt file containing seed points. Mandatory if CDF as input.
+        // Files need time stamp in file name like so: yyyymmdd_hhmmss.txt
+        std::optional<std::filesystem::path> seedPointDirectory [[codegen::directory()]];
+        // Extra variables such as rho, p or t
+        std::optional<std::vector<std::string>> extraVariables;
+        // Which variable in CDF file to trace. b is default for fieldline
```

Comment:
```suggestion

        // Which variable in CDF file to trace. b is default for fieldline
```

Reply:
We should probably remove the default value, since this is no longer just a magnetic fieldline renderable